AmplifyPlugins/Analytics/AWSPinpointAnalyticsPlugin/Dependency/Pinpoint/Endpoint/EndpointClient.swift

```diff
@@ -124,25 +147,21 @@ actor EndpointClient: EndpointClientBehaviour {
             return await configure(endpointProfile: endpointProfile)
         }
 
-        // 2. Look for a valid PinpointEndpointProfile object stored in UserDefaults. It needs to match the current applicationId, otherwise we'll discard it.
-        if let endpointProfileData = userDefaults.data(forKey: Constants.endpointProfileKey),
+        // 2. Look for a valid PinpointEndpointProfile object stored in the Keychain. It needs to match the current applicationId, otherwise we'll discard it.
+        if let endpointProfileData = try? keychain.getData(Constants.endpointProfileKey),
            let decodedEndpointProfile = try? archiver.decode(PinpointEndpointProfile.self, from: endpointProfileData),
            decodedEndpointProfile.applicationId == configuration.appId {
             return await configure(endpointProfile: decodedEndpointProfile)
         }
 
-        userDefaults.removeObject(forKey: Constants.endpointProfileKey)
```

Comment:
You should remove the stored profile from the Keychain here as well, as reaching this line  means it doesn't match the current `applicationId`, so it needs to be discarded.